### PR TITLE
More descriptive error messages for unsupported Jandex storage format versions

### PR DESCRIPTION
**Pull request by Ladicek**

### CODE REVIEW

**Comment by gastaldi** [comment]
Isn't the error message already improved in jandex? I think it's fine to wait for 2.4.1 instead

**Review by gastaldi — APPROVED** [review]
Looks good!

**Comment by Ladicek** [comment]
Right, Jandex has an improved error message now. But Jandex's `IndexReader` doesn't know the problematic file, so can't show it in the error message -- which is what this PR does.

File: core/deployment/src/main/java/io/quarkus/deployment/index/IndexingUtil.java [hunk 1]
```diff
@@ -61,7 +62,11 @@ public static Index indexJar(File file, Set<String> removed) throws IOException
                                 file);
                         return indexJar(jarFile, removed);
                     } else {
-                        return reader.read();
+                        try {
+                            return reader.read();
+                        } catch (UnsupportedVersion e) {
+                            throw new UnsupportedVersion("Can't read Jandex index from " + file + ": " + e.getMessage());
```
Comment (mkouba) [inline core/deployment/src/main/java/io/quarkus/deployment/index/IndexingUtil.java:68]: Hm, this way you swallow the info about supported index versions, or?

Comment (gastaldi) [reply core/deployment/src/main/java/io/quarkus/deployment/index/IndexingUtil.java:68]: I think that info is in the `e.getMessage()`?

Comment (mkouba) [reply core/deployment/src/main/java/io/quarkus/deployment/index/IndexingUtil.java:68]: Ah, of course! Stupid me!!!

**Review by mkouba — APPROVED** [review]

**Pull request merged**